guidance/_grammar.py

```diff
@@ -732,6 +732,35 @@ def string(value) -> Union[str, bytes, Null, Byte, Join]:
         return Join([Byte(b[i:i+1]) for i in range(len(b))], name=str(b))
     
 def select(options: List[_T], name=None, list_append=False, recurse=False, skip_checks=False) -> Union[Select, _T]:
```

Comment:
To match `gen()` this should be changed to
```python
def select(name = None, *, options: List[_T], list_append=False, recurse=False, skip_checks=False) -> Union[Select, _T]:
```
However, that might cause the odd bit of breakage. And I'm not confident our tests are sufficiently comprehensive to catch everything.

Reply:
I don't think the `gen()` convention is the right one to follow here, as required args being forced as keywords feels unintuitive (users shouldn't have to know that `options` is the way we named our list arg in select). I'll think about a better API standardization today but let's not change it for now.

Reply:
OK. Happy to merge without making that change.